File: In_C_Programming/q27.c
#include <stdio.h>

int check_Palindrome(int arr[], int temp[], int ln)
{
    for (int i = 0; i < ln; i++)
    {

        if (arr[i] != temp[i])
        {

            return -1;
        }
    }

    return 1;
}

void reverse_Array(int temp[], int ln)
{
    for (int i = 0; i < ln / 2; i++)
    {
        temp[i] ^= temp[ln - 1 - i];
        temp[ln - 1 - i] ^= temp[i];
        temp[i] ^= temp[ln - 1 - i];
    }
}

void get_Element(int arr[], int temp[], int ln)
{
    printf("Enter The Array Element: \n");

    for (int i = 0; i < ln; i++)
    {
        scanf("%d", &arr[i]);
    }

    for (int i = 0; i < ln; i++)
    {
        temp[i] = arr[i];
    }
}

int main()
{
    int ln;

    printf("Enter The Array Length: ");
    scanf("%d", &ln);

    int arr[ln], temp[ln];

    get_Element(arr, temp, ln);
    reverse_Array(temp, ln);
    int ans = check_Palindrome(arr, temp, ln);

    if (ans == -1)
    {

        printf("This array is not a a Plindrome \n");
    }
    else
    {
        printf("This array is a Palindrome \n");
    }

    return 0;
}

File: In_Java_Programming/q27.java
import java.util.Scanner;

public class q27 {

    public static boolean get_Checked(int arr[], int temp[], int ln) {

        boolean flag = false;

        for (int i = 0; i < ln; i++) {

            if(arr[i] != temp[i]){

                flag = true;
            }
        }

        return flag;
    }

    public static void get_Reversed(int arr[], int ln) {

        for (int i = 0; i < ln / 2; i++) {

            arr[i] ^= arr[ln - 1];
            arr[ln - 1] ^= arr[i];
            arr[i] ^= arr[ln - 1];
        }
    }

    public static void get_Copied(int arr[], int temp[], int ln) {

        for (int i = 0; i < ln; i++) {
            temp[i] = arr[i];
        }
    }

    public static void get_Element(int arr[], int ln, Scanner sc) {

        System.out.println("Enter The Array Elements: ");

        for (int i = 0; i < ln; i++) {

            arr[i] = sc.nextInt();
        }
    }

    public static void main(String[] args) {

        Scanner sc = new Scanner(System.in);

        System.out.println("Enter Array Length: ");
        int ln = sc.nextInt();

        int arr[] = new int[ln];
        int temp[] = new int[ln];

        get_Element(arr, ln, sc);
        get_Copied(arr, temp, ln);
        get_Reversed(arr, ln);
        boolean ans = get_Checked(arr, temp, ln);

        if (ans == false) {
            System.out.println("This array is a Palindrome");
        }else{
            System.out.println("This array is not a Palindrome");
        }

        sc.close();
    }
}